Update activity cancellation docs (#2679)
temporalio/documentation: docs-src/concepts/what-is-an-activity-execution.md

- lets the Activity know it doesn't need to keep doing work, and
- gives the Activity time to clean up any resources it has created.

> Activities must heartbeat to receive cancellations from the server.

Reviewer: ```suggestion
Activities must heartbeat to receive cancellations from a Temporal Service.
```

- In some SDKs:
  - The Worker is shutting down.
  - An Activity sends a Heartbeat but the Heartbeat details can't be converted by the Worker's configured [Data Converter](/concepts/what-is-a-data-converter). This fails the Activity Task Execution with an Application Failure.

>   - The Activity exceeded its attempt timeout on the Worker side and is not heartbeating or the server hasn't relayed its timeout cancellation.

Reviewer: ```suggestion
  - The Activity timed out on the Worker side and is not Heartbeating or the Temporal Service hasn't relayed a Cancellation.
```

This sentence doesn't make any sense to me. Suggesting something which is possibly what it is meant to say?
